cli/command/stack/list.go

```diff
@@ -28,6 +28,9 @@ func newListCommand(dockerCli command.Cli) *cobra.Command {
 
 	flags := cmd.Flags()
 	flags.StringVar(&opts.Format, "format", "", "Pretty-print stacks using a Go template")
+	flags.StringSliceVar(&opts.Namespaces, "namespace", []string{}, "Kubernetes namespaces to use")
```

Comment:
All ```docker stack``` sub-commands use ```AddNamespaceFlag``` and have a ```namespace``` flag defined as a ```StringVar``` except ```ls``` which has a ```namespace``` flag defined as a ```StringSliceVar```.
This is because for all sub-commands but ```ls``` only one ```--namespace``` can be set whereas ```ls``` can have several, e.g. ```docker stack ls --namespace=tutu,default,bla``` or ```docker stack ls --namespace=tutu --namespace=default --namespace=bla```.

Reply:
Wondering now if this should be a `StringSliceVar` or a `StringVar`; I think `StringVar` can be specified multiple times (`--namespace=bla --namespace=foo`), but does not support comma-separated values; we haven't documented using comma-separated values, and don't think we support it for other  options (possibly `--security-opt` only)

(note that if a command only accepts a single namespace, we could still produce an error if multiple are given)

Reply:
I'll take another look but from what I remember there was no way to make it look nice, for both case, e.g.
```
$ docker stack ps  --help
…
Options:
…
      --namespace string    Kubernetes namespace to use
```
VS
```
$ ./build/docker.exe stack ls --help
…
Options:
…
      --namespace strings   Kubernetes namespaces to use
```
Also I don't see how a single `StringVar` can hold multiple `--namespace` values…

Reply:
> I think StringVar can be specified multiple times (--namespace=bla --namespace=foo), but does not support comma-separated values
In that case the latest value is silently selected (here `foo`), I suppose the flags are simply processed in order.
If we just let comma-separated values out of the question we could indeed emulate both behaviors with a `StringSliceVar` by using the last element when only one namespace is required.

(Incidentally passing `--namespace=bla,foo` to a `StringVar` flag yields a value of `bla,foo`…)

Reply:
The best I could come up with by using `[]string` on the `stack` command level is
```
$ ./build/docker.exe stack --help

Usage:  docker stack COMMAND

Manage Docker stacks

Options:
      --kubeconfig string   Kubernetes config file
      --namespace strings   Kubernetes namespace to use
```
This is wrong for all sub-commands as for most of them it says `strings` plural whereas only one is supported (and differs for instance from `kubeconfig`), and for `docker stack ls` which supports several namespaces the description should have `namespaces`…

I don't see how to have types and descriptions accurate for all sub-commands without duplicating the flag.